applications/annotate_points.md

```diff
@@ -0,0 +1,314 @@
+# annotating videos with napari
+
+In this tutorial, we will use napari (requires version 0.3.2 or greater) to make a simple GUI application for annotating points in videos. This GUI could be useful for making annotations required to train algorithms for markless tracking of animals (e.g., [DeepLabCut](http://www.mousemotorlab.org/deeplabcut)). In this tutorial, we will cover creating and interacting with a Points layer with properties (i.e., labels for the points), connecting custom UI elements to events, and creating custom keybindings.
+
+At the end of this tutorial, we will have created a GUI for annotating points in videos that we can simply call by:
+
+```python
+im_path = '<path to directory with data>/*.png'
+output = '<path to directory with data>/annotations.csv'
+point_annotator(im_path, labels=['ear_l', 'ear_r', 'tail'], output_path=output)
+
+```
+that looks like this:
+
+![image]({{ '/assets/tutorials/point_annotator_demo.gif' | relative_url }})
+
+
+You can explore the project in [this repository](https://github.com/kevinyamauchi/PointAnnotator) or check out the main function below. We will walk through the code in the following sections.
+
+```python
+def point_annotator(
+        im_path: str,
+        labels: List[str],
+):
+    """Create a GUI for annotating points in a series of images.
+
+    Parameters
+    ----------
+    im_path : str
+        glob-like string for the images to be labeled.
+    labels : List[str]
+        list of the labels for each keypoint to be annotated (e.g., the body parts to be labeled).
+    """
+    stack = imread(im_path)
+    
+    with napari.gui_qt():
+        viewer = napari.view_image(stack, contrast_limits=[0, 256])
+        properties = {'label': labels}
+        points_layer = viewer.add_points(
+            properties=properties,
+            edge_color='label',
+            edge_color_cycle=COLOR_CYCLE,
+            symbol='o',
+            face_color='transparent',
+            edge_width=8,
+            size=12
+        )
+        points_layer.edge_color_mode = 'cycle'
+
+        # Create the label selection menu
+        @magicgui(label={'choices': labels})
+        def label_selection(label):
+            return label
+        label_menu = label_selection.Gui()
+
+        def update_label_menu(event):
+            """Update the label menu when the point selection changes"""
+            label_menu.label = points_layer.current_properties['label'][0]
+
+        points_layer.events.current_properties.connect(update_label_menu)
+
+        def label_changed(result):
+            """Update the Points layer when the label menu selection changes"""
+            selected_label = result
+            current_properties = points_layer.current_properties
+            current_properties['label'] = np.asarray([selected_label])
+            points_layer.current_properties = current_properties
+
+        label_menu.label_changed.connect(label_changed)
```

Comment:
I'd pull all of this out into its own function, `create_label_menu`, that takes a points layer as an input. I think the amount of code inside the `with napari.gui_qt()` block is huge, so it'd be good to reduce it into manageable chunks.

Reply:
Good idea! Done.